oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java

```diff
@@ -104,6 +106,14 @@ public static ElasticBulkProcessorHandler getBulkProcessorHandler(@NotNull Elast
         PropertyState async = indexDefinition.getDefinitionNodeState().getProperty("async");
 
         if (async != null) {
+            Iterable<String> opt = async.getValue(Type.STRINGS);
+
+            // Check if this indexing call is a part of async cycle or a commit hook
+            // In case it's from async cycle - commit info will have a indexingCheckpointTime key.
+            // Otherwise it's part of commit hook based indexing due to async property having a value nrt
+            if (!commitInfo.getInfo().containsKey(IndexConstants.CHECKPOINT_CREATION_TIME)) {
```

Comment:
we skip ES acknowledgment for every `async` lane. Should we skip it only when the async configuration is `[async, nrt]`?

Reply:
no - we are skipping ES ack when the async property is set but the commit info does not have the checkpoint creation time in it, this would be the case where the index update is called via the commit hook (due to nrt).

When the index update is called via the async cycle commitInfo.getInfo().containsKey(IndexConstants.CHECKPOINT_CREATION_TIME) would be true.

Reply:
Got it. It sounds good.